src/Status.cxx

```diff
@@ -115,48 +71,55 @@ void FileWriter::Status::MessageInfo::reset() {
 }
 
 std::pair<double, double> FileWriter::Status::MessageInfo::getMbytes() const {
-  return std::pair<double, double>{Mbytes.load(), MbytesSquare.load()};
+  return std::pair<double, double>{Mbytes, MbytesSquare};
 }
 
 std::pair<double, double> FileWriter::Status::MessageInfo::getMessages() const {
-  return std::pair<double, double>{Messages.load(), MessagesSquare.load()};
+  return std::pair<double, double>{Messages, MessagesSquare};
 }
 
-double FileWriter::Status::MessageInfo::getErrors() const {
-  return Errors.load();
+double FileWriter::Status::MessageInfo::getErrors() const { return Errors; }
+
+std::pair<double, double> &operator+=(std::pair<double, double> &First,
+                                      const std::pair<double, double> &Second) {
+  First.first += Second.first;
+  First.second += Second.second;
+  return First;
 }
 
-////////////////////////
-// StreamMasterInfo
 void FileWriter::Status::StreamMasterInfo::add(
-    const std::string &topic, FileWriter::Status::MessageInfo &info) {
-  // The lock prevents Streamer::write from update info while
-  // collecting these stats, or this function from reset while/after
-  // Streamer::write updates. Pairs with MessageInfo::message(error) lock
-  if (StreamsInfo.find(topic) != StreamsInfo.end()) {
-    //    std::lock_guard<std::mutex> lock(info.getMutex());
-    StreamsInfo[topic] = info;
-    Total += info;
-    info.reset();
-  } else {
-    //    std::lock_guard<std::mutex> lock(info.getMutex());
-    StreamsInfo[topic] += info;
-    Total += info;
-    info.reset();
-  }
+    FileWriter::Status::MessageInfo &Info) {
+  Mbytes += Info.getMbytes();
+  Messages += Info.getMessages();
+  Errors += Info.getErrors();
+  Info.reset();
 }
 
 void FileWriter::Status::StreamMasterInfo::setTimeToNextMessage(
     const std::chrono::milliseconds &ToNextMessage) {
-  NextMessageRelativeEta = ToNextMessage;
+  MillisecondsToNextMessage = ToNextMessage;
 }
 const std::chrono::milliseconds
 FileWriter::Status::StreamMasterInfo::getTimeToNextMessage() {
-  return NextMessageRelativeEta;
+  return MillisecondsToNextMessage;
 }
 const std::chrono::milliseconds
 FileWriter::Status::StreamMasterInfo::runTime() {
   auto result = std::chrono::duration_cast<std::chrono::milliseconds>(
       std::chrono::system_clock::now() - StartTime);
   return result;
 }
+
+std::pair<double, double>
+FileWriter::Status::StreamMasterInfo::getMbytes() const {
+  return Mbytes;
+}
+
+std::pair<double, double>
+FileWriter::Status::StreamMasterInfo::getMessages() const {
+  return Messages;
+}
+
+double FileWriter::Status::StreamMasterInfo::getErrors() const {
+  return Errors;
+}
```

Comment:
Missing new line at end of file

Reply:
Can you please give a look at the file as it is now? I tried to force the new line, but in that case the build failed on Jenkins due to clang-format (as you can see in the previous commit that failed)

Reply:
I'm not sure what happened, but it looks correct now.